libsrc/sym1/tapeio.s

```diff
@@ -0,0 +1,47 @@
+;
+; Wayne Parham (wayne@parhamdata.com)
+;
+; int loadt (int id);
+; int dumpt (int id, int start_addr, int end_addr);
+;
+
+.include        "sym1.inc"
+
+.import         popax, return0, return1
+
+.export         _loadt, _dumpt
+
+.segment        "CODE"
+
+.proc _loadt:   near
+
+        sta     P1L              ; Tape record ID to P1L
+        ldx     #$00
+        stx     P1H
+        ldy     #$80
+        jsr     LOADT            ; Read data from tape
+        bcs     error
+        jmp     return0          ; Return 0 if sucessful
+error:  jmp     return1          ; or 1 if not
+
+.endproc
+
+.proc _dumpt:  near
+
+        sta     P3L              ; End address
+        stx     P3H
+        jsr     popax
+        sta     P2L              ; Start address
+        stx     P2H
+        jsr     popax
+        sta     P1L              ; Tape Record ID
+        ldx     #$00
```

Comment:
When this function has an `int` parameter, meaning 16 bit and you set `P1` which seems to be a 16 bit value, why do you then trash the high byte by setting X to zero?

Reply:
Updated to cast "ID" as unsigned char rather than int.